table/block_based/block_based_table_builder.cc

```diff
@@ -70,7 +70,8 @@ FilterBlockBuilder* CreateFilterBlockBuilder(
     const ImmutableCFOptions& /*opt*/, const MutableCFOptions& mopt,
     const FilterBuildingContext& context,
     const bool use_delta_encoding_for_index_values,
-    PartitionedIndexBuilder* const p_index_builder) {
+    PartitionedIndexBuilder* const p_index_builder, size_t ts_sz,
+    const bool persist_user_defined_timestamps) {
```

Comment:
I'm wondering if we need these new parameters/members (not just here but in other classes) - seems to me in many cases the class already has access to both the timestamp size and the "persist timestamps" flag (via the comparator and/or the options structure).

Reply:
There is indeed some passing around of these two flags which seem redundant. Mainly because`IndexBuilder` has access to the comparator but not the options structure, `PartitionedFilterBlockBuilder` does not have access to these two. And since there is one dup step of getting the user comparator and then getting the timestamp size. I think it may be more straightforward for the `BlockBasedTableBuilder::Rep` to remember these two things and every class below it to use that as source of truth. Later when handling the value part for RangeDeletion entries, it can directly use flags in `Builder::Rep` too.

Reply:
Sounds good!